### PR TITLE
Test if DNS service is listening on local port 53 when reporting status

### LINKED ISSUE #1050: Proper reporting of Pi-hole status
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*
- [X] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:** 10

---

INTERNAL

**[BUG | ISSUE] Expected Behaviour:**

Pi-hole is shown as `disabled` or `failure` if `dnsmasq` is not running at all (e.g. due to being OOM killed)

**[BUG | ISSUE] Actual Behaviour:**

`pihole status` only investigates whether the line `addn-hosts=` is enabled or not in the config file. No status check of `dnsmasq` process.

Web interface shows `Active` even if `dnsmasq` is not even started, hence DNS resolution does not work at all

**[BUG | ISSUE] Steps to reproduce:**

- Stop `dnsmasq` service.

or

- Insert illegal parameter into config file to prevent starting of `dnsmasq` via web interface permanently.

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 10

---
Fixes #1050 

Test if DNS service is listening on local port 53 when reporting status. 

Current behavior:
```
root@ubuntu-512mb-fra1-01:/etc/.pihole# service dnsmasq stop
root@ubuntu-512mb-fra1-01:/etc/.pihole# pihole status
::: Pi-hole blocking is Enabled
root@ubuntu-512mb-fra1-01:/etc/.pihole# service dnsmasq start
root@ubuntu-512mb-fra1-01:/etc/.pihole# pihole status
::: Pi-hole blocking is Enabled
```

Proposed behavior:
```
root@ubuntu-512mb-fra1-01:/etc/.pihole# service dnsmasq stop
root@ubuntu-512mb-fra1-01:/etc/.pihole# pihole status
::: DNS service is NOT running
root@ubuntu-512mb-fra1-01:/etc/.pihole# service dnsmasq start
root@ubuntu-512mb-fra1-01:/etc/.pihole# pihole status
::: DNS service is running
::: Pi-hole blocking is Enabled
```

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

**Comment by thomasleveil** [comment]
This tests port 53/tcp, shouldn't it also test port 53/udp?

File: pihole [hunk 1]
```diff
@@ -185,6 +185,19 @@ piholeLogging() {
 }
 
 piholeStatus() {
+  if [[ $(netstat -plnt | grep -c ':53') > 0 ]]; then
```
Comment (thomasleveil) [inline pihole]: by adding a space after `:53` we make sure grep will exclude ports starting with `53`. On my system :

```
$ netstat -plnu | grep -c ':53'
udp        0      0 0.0.0.0:5317            0.0.0.0:*                           -
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           -
udp6       0      0 :::53                   :::*                                -
udp6       0      0 :::5353                 :::*                                -
```
while 

```
$ netstat -plnu | grep ':53 '
udp6       0      0 :::53                   :::*                                -
```

**Comment by dschaper** [comment]
`dnsmasq` will open both UDP and TCP when started. Since DNS needs TCP for fallback (big query or zone transfer or DNSSEC) it's probably overkill to check for both. (And I'm going off memory, but I think that even if UDP 53 is off/blocked, clients will failover to TCP and not stall out.)

**Comment by DL6ER** [comment]
> if UDP 53 is off/blocked, clients will failover to TCP

I think so, too. It is just a simple test I came up with which should work independent of the OS.

**Comment by dschaper** [comment]
Approved

**Pull request merged**